### PR TITLE
Module2 HW

### PR DESCRIPTION
**Pull request by viktordavidovich**

### CODE REVIEW

File: module2/server.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+    if (user) {
+        res.send(user);
+    } else {
+        res.send('No user found.');
+    }
+
+});
+
+app.post('/users', (req, res) => {
```
Comment (gpukys) [inline module2/server.ts:56]: This should be a GET request. According to the standards

File: module2/server.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+    const loginSubstring: string = req.body.login || '';
+    const limit: number = req.body.limit;
```
Comment (gpukys) [inline module2/server.ts:59]: This should then use query params instead of body

File: module2/server.ts [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+    const allUsers = users.getAllUsers();
+    let limitUsers: UserModel[] = [];
+
+    allUsers.forEach((user) => {
```
Comment (gpukys) [inline module2/server.ts:64]: Check if loginSubstring is empty before looping through

File: module2/server.ts [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+            if (nameA < nameB) {
+                return -1;
+            }
+            if (nameA > nameB) {
+                return 1;
+            }
+
+            return 0;
+        }).slice(0, limit);
```
Comment (gpukys) [inline module2/server.ts:83]: What if limit was undefined or a string like aaa?

File: module2/server.ts [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+
+    users.addUser(user);
+    res.send(users.getUser(user.id));
+});
+
+app.post('/user/:id', (req, res) => {
+
+    let user: UserModel = users.getUser(req.params.id);
+
+    if (req.body.login !== user.login && req.body.password !== user.password) {
```
Comment (gpukys) [inline module2/server.ts:114]: This should use an OR operator

File: module2/server.ts [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+        return res.status(400).send({
+            message: "login & password do not match our records."
+        });
+    }
+
+    user.login = req.body.login;
+    user.password = req.body.password;
+    user.age = req.body.age;
+
+    users.addUser(user);
```
Comment (gpukys) [inline module2/server.ts:125]: Confusing. This is updating a user, not adding one

File: module2/server.ts [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+
+    res.send(users.getUser(user.id));
+});
+
+
+app.delete('/user/:id', (req, res) => {
+
+    let user: UserModel = users.getUser(req.params.id);
+    user.isDeleted = true;
```
Comment (gpukys) [inline module2/server.ts:134]: what if no user has been found?

File: module2/server.ts [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+            return 0;
+        }).slice(0, limit);
+
+        res.send(limitUsers);
+    } else {
+        res.send('No users found.');
+    }
+});
+
+app.put('/user', (req, res) => {
```
Comment (gpukys) [inline module2/server.ts:91]: This should use POST according to standards

File: module2/server.ts [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+        isDeleted: false
+    };
+    const validate = ajv.getSchema<UserModel>("user");
+
+    if (!validate(user)) {
+        return res.status(400).send(combineErrors(validate.errors));
+    }
+
+    users.addUser(user);
+    res.send(users.getUser(user.id));
```
Comment (gpukys) [inline module2/server.ts:107]: Why not just send ```user```?

File: module2/server.ts (hunk 5 above)
Comment (gpukys) [inline module2/server.ts:114]: What if there is no user?

**Review by viktordavidovich — COMMENTED** [review]
All comments has been fixed.

**Pull request merged**